README.md

```diff
@@ -3403,6 +3403,16 @@ Other Style Guides
     - We primarily use [`mocha`](https://www.npmjs.com/package/mocha) at Airbnb. [`tape`](https://www.npmjs.com/package/tape) is also used occasionally for small, separate modules.
     - 100% test coverage is a good goal to strive for, even if it’s not always practical to reach it.
     - Whenever you fix a bug, _write a regression test_. A bug fixed without a regression test is almost certainly going to break again in the future.
+    
+  <a name="testing--is-documentation"></a>
+  - [30.3](#testing--is-documentation) **Is your documentation**:
+    - Tests should explain what your code does.
+    - First write a case for each main functionality, and then their exceptions.
+    - Refactor tests until they are easy to read and understand.
+    - Never use _"private"_ properties to test code.
+    - Code inside tests should be copy-pasteable code to make new developments.
+    - If some functionality works but is not tested, do not use it. It may break your code in the future.
```

Comment:
I'd say if it's not tested, then it objectively doesn't work, because you don't have a test proving it does :-)

Reply:
Uhmm, yeah!

What I have found that some times there are some behaviours that emerge from the current code structure, for example:

```javascript
getCar(carIndex) {
   return carList[carIndex];
}
```

It turns out that if you call to `getCar(undefined)` it returns undefined.
Some people think that to return undefined if argument undefined it is a desired functionality, 
and they start to put inputs to `getCar` that may have an undefined.

Some time later, you may decide to do something else with carIndex that will break if it is undefined. 
And then you will break previous code.
And... if they did a lot of mocking, you will not discover it until it is too late (or difficult to trace).

In this scenario coders can say that the function is tested. But if there was no case in which it returns undefined, they never should expected and undefined.

Of course you can add 1.000 cases for each combination, but 1: it is not feasible, 2: when you define an API you define which cases are accepted. 

So at the end, specially you should look test code always before using something in some way that was not designed for.

What about:

    - If it is not tested, than it objectively does not work, 
      because you do not have a test proving it does.
      Including those cases that are not present.

Reply:
Sounds good to me.